From ed42f47575284a7d67f4128334092787aff002ec Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Wed, 8 Aug 2018 11:12:28 -0700
Subject: [PATCH 2071/2367] ASoC: SOF: do not add virtual routes to route list

Skip adding virtual routes to the route list. This will prevent
sending the routes to the FW during system resume.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index 4f2e9cf..702d8a2 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -1957,6 +1957,8 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 	    sink_swidget->id != snd_soc_dapm_buffer) {
 		dev_dbg(sdev->dev, "warning: neither Linked source component %s nor sink component %s is of buffer type, ignoring link\n",
 			route->source, route->sink);
+		ret = 0;
+		goto err;
 	} else {
 		ret = sof_ipc_tx_message(sdev->ipc,
 					 connect->hdr.cmd,
@@ -1981,13 +1983,13 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 			ret = reply.error;
 			goto err;
 		}
-	}
 
-	sroute->route = route;
-	sroute->private = connect;
+		sroute->route = route;
+		sroute->private = connect;
 
-	/* add route to route list */
-	list_add(&sroute->list, &sdev->route_list);
+		/* add route to route list */
+		list_add(&sroute->list, &sdev->route_list);
+	}
 
 	return ret;
 
-- 
2.7.4

